src/daemon/index.ts

```diff
@@ -319,6 +324,86 @@ export function sendLSPUsageStats() {
    }
 }
 
+class HybridLSPStats {
+   private lspStats: LSPUsageStats; // standard lsp stats
+   private ssLspStats: LSPUsageStats | undefined; // syntax server lsp stats
+
+   public constructor(readonly javaExtVersion: string, readonly sampling: string) {
+      this.lspStats = new LSPUsageStats(javaExtVersion, sampling, false);
+      // These pre-release versions include the fromSyntaxServer property in the requestEnd event,
+      // but not in the requestStart event.
+      const ignoreVersions = [
+         "1.24.2023100604",
+         "1.24.2023100504",
+         "1.24.2023100404",
+      ];
+      if (sampling !== "pre-release" || !ignoreVersions.includes(javaExtVersion)) {
```

Comment:
Add a `TODO` marker here to update the logic before stable release? (if I understand correctly, the code here is to temporarily collect syntax server traces via pre-release channel)

Reply:
done.